lesson3/code-examples/examples/exercise.js

```diff
@@ -0,0 +1,172 @@
+  ) {
+    return "Leo";
+  } else if (
+    (month === 8 && day >= 23 && day <= 31) ||
+    (month === 9 && day >= 1 && day < 23)
+  ) {
+    return "Virgo";
+  } else if (
+    (month === 9 && day >= 23 && day <= 30) ||
+    (month === 10 && day >= 1 && day < 23)
+  ) {
+    return "Libra";
+  } else if (
+    (month === 10 && day >= 23 && day <= 31) ||
+    (month === 11 && day >= 1 && day < 22)
+  ) {
+    return "Scorpio";
+  } else if (
+    (month === 11 && day >= 22 && day <= 31) ||
+    (month === 12 && day >= 1 && day < 22)
+  ) {
+    return "Sagittarius";
+  } else if (
+    (month === 12 && day >= 22 && day <= 30) ||
+    (month === 1 && day >= 1 && day < 20)
+  ) {
+    return "Capricorn";
+  } else {
+    return "Zodiac sign for given date not exists!";
+  }
+}
+
+// 2. Take data from file users.json and output person names with salary that are bigger than 1000. Try to use *Array* object methods like *forEach*.
+
+function getSalary(url) {
+  return fetch(url)
+    .then(res => {
+      if (!res.ok) {
+        throw new Error(`Erron occured ${res.status}`);
+      }
+      return res.json();
+    })
+    .then(data => {
+      data.forEach(user => {
+        user.salary > 1000 ? console.log(user.name) : "";
+      });
+    }).catch(err => console.log('Error message:', err));
+}
+
+getSalary("../users.json");
+
+// 3. Print current bitcoin price using https://api.gdax.com/products/BTC-USD/book
+
+function getBitcoinValue(url) {
+  return fetch(url)
+    .then(res => {
+      if (!res.ok) {
+        throw new Error(`Erron occured ${res.status}`);
+      }
+      return res.json();
+    })
+    .then(data => {
+      console.log(`Current Bitcoin price: ${data.asks[0][0]} Usd`);
+    }).catch(err => console.log('Error message:', err));
+}
+
+getBitcoinValue("https://api.gdax.com/products/BTC-USD/book");
+
+//    More advanced. Optional:
+//4. Take data from file *users.json* and create Users array. Sort this array by users name. Try to use *Array sort* method.
+
+function sortUsers(url) {
+	return fetch(url)
+    .then(res => {
+      if (!res.ok) {
+        throw new Error(`Erron occured ${res.status}`);
+      }
+      return res.json();
+    })
+    .then(data => {
+			const users = data.map(user => {
+				return user.name
+			})
+			const sorted = users.sort();
+			console.log(sorted);
+      
+    }).catch(err => console.log('Error message:', err));
+}
+
+sortUsers("../users.json");
+
+//5. Take data from file *users.json* and create Users array. Filter this array so output will contains only that users that have salary bigger than 1000. Try to use *Array filter* method.
+//
+
+function filterSalary(url) {
+  return fetch(url)
+    .then(res => {
+      if (!res.ok) {
+        throw new Error(`Erron occured ${res.status}`);
+      }
+      return res.json();
+    })
+    .then(data => {
+      const users = data.filter(user => {
+        return user.salary > 1000;
+      });
+			users.map(user => console.log(user.name));
```

Comment:
`map` is generally used to create a new array without impacting the original array. If you only want to do some action with each array element (in this case output values) then it's better to use `foreach`.